src/foam/nanos/logger/RepeatLogger.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+
+foam.CLASS({
+  package: 'foam.nanos.logger',
+  name: 'RepeatLogger',
+  extends: 'foam.nanos.logger.ProxyLogger',
+  requires: [
+    'foam.nanos.logger.LogLevel'
+  ],
+
+  properties: [
+    {
+      class: 'Object',
+      name: 'lastUniqueObject',
+      value: null
+    },
+    {
+      class: 'Int',
+      name: 'repeatCount',
+      value: 0
+    },
+    {
+      class: 'Enum',
+      of: 'foam.nanos.logger.LogLevel',
+      name: 'lastLogLevel',
+      factory: function() { return foam.nanos.logger.LogLevel.INFO; }
+    }
+  ],
+
+  methods: [
+    {
+      name: 'newLogHandler',
+      args: [ 
+        {
+          name: 'logLevelName',
+          javaType: 'Enum'
+        },
+        {
+          name: 'setLastLogLevel',
+          javaType: 'Boolean'
+        },
+        {
+          name: 'args',
+          javaType: 'Object...'
+        },
+      ],
+      javaReturns: 'void',
+      javaCode: `
+  switch ( LogLevel.values()[logLevelName.ordinal()] ) {
+    case DEBUG: {
+      setRepeatCount(1);
+      setLastUniqueObject(args);
+      getDelegate().debug(args);  
+      if (setLastLogLevel){
+        setLastLogLevel(LogLevel.DEBUG);
+      }
+      break;
+    }
+    case INFO: {
+      setRepeatCount(1);
+      setLastUniqueObject(args);
+      getDelegate().info(args);  
+      if (setLastLogLevel){
+        setLastLogLevel(LogLevel.INFO);
+      }
+      break;
+    }
+    case WARNING: {
+      setRepeatCount(1);
+      setLastUniqueObject(args);
+      getDelegate().warning(args);  
+      if (setLastLogLevel){
+        setLastLogLevel(LogLevel.WARNING);
+      }
+      break;
+      
+    }     
+    case ERROR: {
+      setRepeatCount(1);
+      setLastUniqueObject(args);
+      getDelegate().error(args);  
+      if (setLastLogLevel){
+        setLastLogLevel(LogLevel.ERROR);
+      }
+      break;
+    }
+  }
+`  
+    },
+    {
+      name: 'logLastLogRepeats',
+      javaReturns: 'void',
+      javaCode: 
+      `
+  switch ( getLastLogLevel() ) {
+    case DEBUG: {
+      getDelegate().debug("The last log was repeated " + getRepeatCount() + " times" );
+      break;
+    }
+    case INFO: {
+      getDelegate().info("The last log was repeated "+ getRepeatCount()  + " times" );
+      break;
+    }
+    case WARNING: {
+      getDelegate().warning("The last log was repeated "+ getRepeatCount()  + " times" );
+      break;      
+    }        
+    case ERROR: {
+      getDelegate().error("The last log was repeated " + getRepeatCount() + " times" );
+      break;
+    }
+  }     
+`
+    },
+    {
+      name: 'repeatLogConstructor',
```

Comment:
This is an odd name, since it isn't an actually a constructor. Perhaps a different name would be better.